Migrate wiki articles to Vaadin documentation

eriklumme opened the pull request:
Using `contents.asciidoc` from hezamu@32a651d5eb4ecf7567e91df3437e204be43fc6e5

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/9958)
<!-- Reviewable:end -->

Commit:
Migrate UsingURIFragments

Commit:
Migrate UsingURIFragments

Commit:
Add missing changes

Commit:
Add missing changes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/framework?pullRequest=9958) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/framework?pullRequest=9958) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/framework?pullRequest=9958) it.</sub>

hezamu committed:
Migrate wiki articles to Vaadin documentation (#9912)

* Vaadin Tutorial For Swing Developers
* Setting And Reading Session Attributes
* Enabling Server Push
* Cleaning Up Resources In A UI
* Sending Email From Java Applications
* Using Parameters With Views
* Optimizing Sluggish UI
* Configuring Push For Your Enviroment
* Setting And Reading Cookies
* Using Polling
* Creating An Application That Preserves State On Refresh
* Finding The Current UI And Page And Vaadin Session
* Sending Events From The Client To The Server Using RPC
* Handling Logout
* Remember To Set The Locale
* Scalable Web Applications
* MVC Basics In ITMill Toolkit
* Access Control For Views
* Customizing The Startup Page In An Application

wbadam committed:
Remove element explicitly when reordering tree grid rows (#9952)

Under certain circumstances IE 11 (11.0.45 / 11.0.9600.18762) produces an exception when collapsing/expanding rows (particularly the first child after the very first element) in a TreeGrid within a Window.

This workaround removes the row explicitly before inserting, instead of letting JS handle it.

Fixes #9850

Commit:
Migrate AccessingWebPageAndBrowserInformation

Commit:
Migrate AccessingWebPageAndBrowserInformation

hesara commented:
Reviewed 2 of 2 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9958#-KtkEiNSWbaISo_wqG2J:-KtkEiNSWbaISo_wqG2K:bq0dx4i) (no related file):*
Please rebase this on top of latest master where hezamu's changes are included. I'll probably keep reviewing this PR as commits come in, but not merging until there is a longer calm period to reduce git merge overhead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9958)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate ConfiguringPushForYourEnvironment

Commit:
Migrate ConfiguringPushForYourEnvironment

hesara commented:
It seems to me that ConfiguringPush... was already there.

Commit:
Merge branch 'wiki-articles' of https://github.com/eriklumme/framework into wiki-articles

eriklumme commented:
Indeed, although it wasn't marked as such in the spreadsheet. I merged it with mine, there were a couple dead links that I had updated.

Commit:
Fix typo

Commit:
Fix typo
